src/main/frontend/fs/sync.cljs

```diff
@@ -1539,6 +1561,63 @@
                            delete-filetxns)]
     (set (concat update-file-items rename-file-items delete-file-items))))
 
+
+(defn- <fetch-remote-and-update-local-files
+  [graph-uuid base-path relative-paths]
+  (go
+    (let [fetched-file-rpaths (<! (<fetch-remote-files rsapi graph-uuid base-path relative-paths))]
+      (p->c (p/all (->> fetched-file-rpaths
+                        (map (fn [rpath]
+                               (p/let [incoming-file (path/path-join "logseq/version-files/incoming" rpath)
+                                       base-file (path/path-join "logseq/version-files/base" rpath)
+                                       current-change-file rpath
+                                       format (gp-util/get-format current-change-file)
+                                       repo (state/get-current-repo)
+                                       repo-dir (config/get-repo-dir repo)
+                                       base-exists? (fs/file-exists? repo-dir base-file)]
+                                 (cond
+                                   base-exists?
+                                   (p/let [base-content (fs/read-file repo-dir base-file)
+                                           current-content (fs/read-file repo-dir current-change-file)]
+                                     (if (= base-content current-content)
+                                       (do
+                                         (prn "base=current, write directly")
+                                         (p/do!
+                                          (fs/copy! repo
+                                                    (path/path-join repo-dir incoming-file)
+                                                    (path/path-join repo-dir current-change-file))
+                                          (fs/copy! repo
+                                                   (path/path-join repo-dir incoming-file)
+                                                   (path/path-join repo-dir base-file))))
+                                       (do
+                                         (prn "base!=current, should do a 3-way merge")
+                                         (p/let [incoming-content (fs/read-file repo-dir incoming-file)
+                                                 merged-content (diff-merge/three-way-merge base-content incoming-content current-content format)]
+                                           (prn ::merged-content merged-content)
+                                           (when (seq merged-content)
```

Comment:
will there be a situation where incoming-content is not empty & merged-content is empty?